### PR TITLE
Abstracting link components by reexporting them.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is creating an abstraction around our `react-router`-specific
link components (`Link` & `LinkContainer`) by reexporting them from an
own module. This allows us to replace them easily across core & plugins
when we are replacing `react-router` or `react-router-bootstrap` in a
future step.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/events/events/Events.jsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import lodash from 'lodash';
-import { Link } from 'react-router';
-import { LinkContainer } from 'react-router-bootstrap';
 import styled, { css } from 'styled-components';
 
+import { Link, LinkContainer } from 'components/graylog/router';
```
Comment (Gaya) [inline graylog2-web-interface/src/components/events/events/Events.jsx:6]: Right on! Saved a line of code 🤓

File: graylog2-web-interface/src/views/components/views/MissingRequirements.jsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // @flow strict
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-// $FlowFixMe: imports from core need to be fixed in flow
-import { LinkContainer } from 'react-router-bootstrap';
 
+// $FlowFixMe: imports from core need to be fixed in flow
```
Comment (Gaya) [inline graylog2-web-interface/src/views/components/views/MissingRequirements.jsx:5]: Are these FlowFixMe still needed?

**Review by Gaya — APPROVED** [review]
This approach looks safe and good to me. I'd do the same thing.

Is there a way to ignore the weblinter on jenkins since you're touching this many old files?

File: graylog2-web-interface/src/views/components/views/MissingRequirements.jsx (hunk 2 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/views/components/views/MissingRequirements.jsx:5]: No, good catch!

**Comment by dennisoelkers** [comment]
@Gaya: We _could_ introduce a flag for Jenkins to skip the linter job for jobs like this, or backports where we want to keep the amount of changes minimal. But I am not sure if this would be good, because this way the reviewer has to look and think twice if the linter job is red. You can still merge the PR if you have the same opinion for the linter job to be ok to fail. Does that make sense?

**Comment by Gaya** [comment]
> @Gaya: We _could_ introduce a flag for Jenkins to skip the linter job for jobs like this, or backports where we want to keep the amount of changes minimal. But I am not sure if this would be good, because this way the reviewer has to look and think twice if the linter job is red. You can still merge the PR if you have the same opinion for the linter job to be ok to fail. Does that make sense?

Can we already merge even though the linter is red? I have mixed feelings about that, but it would be the same as putting in a flag to disable linting.

But yeah, I am okay with being able to merge even though the linter says no, maybe it's up to the person creating the PR to put in a comment that it's only touching minor parts of old files and fixing the linting errors makes no sense?

**Comment by dennisoelkers** [comment]
> Can we already merge even though the linter is red?

Yes, that should work.

> [...] maybe it's up to the person creating the PR to put in a comment that it's only touching minor parts of old files and fixing the linting errors makes no sense?

I would think that the author should add a comment if it is not obvious. There might be cases where the author thinks it is obvious, while the reviewer does not, but that would just mean that the reviewer might ask if the linter is red on purpose or not.

**Comment by Gaya** [comment]
100% agree with this methodology.

**Pull request merged**